### DISCUSSION TITLE
Welcome to roborock-oucher Discussions!

### DISCUSSION THREAD

**Discussion by ale-rinaldi** [porech/roborock-oucher#31] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by charliesz** [comment]
Hello everyone. I'm late to the party. I have a couple of roborosck S7 and S5 robots at home with valetudo. I was wondering if I could use the oucher project in some way to let my robot speak some phrases triggert in a way from homeassistant. Like "the wasching maschine is finshed". Or some stupid idea: Let the robo speak from the lateste computer game 7days: "Master, we're not alone" if someone is entering the house.

If I got it correct this script just watches the log and plays the sound. Maybe it is possible to get some log entry like: reset "dock bin", set "carpet mode off/on" or reset "main brush". I think you can trigger that via mqtt but I did not investigate the log entrys on the robot.

Maybe someone has already done something similiar.

**Reply by ale-rinaldi** [reply]
Hi there! Nice idea, but it's probably just worth doing another project that just subscribes to MQTT and plays sounds based on received messages, without even interfering with the logs.
At this point, it would have nothing specific to the Roborock, it may work on any PC but, at the end of the day, the Roborock is just a PC as another with some fancy hardware connected...

**Reply by charliesz** [reply]
Thanks for your input. Your idea is a lot more straight forward. I hope I can give it a shot during the weekend. I will report back, if I got it working.

**Reply by ale-rinaldi** [reply]
> Thanks for your input. Your idea is a lot more straight forward. I hope I can give it a shot during the weekend. I will report back, if I got it working.

Yeah, let me know how it goes! You may start from this project since it already has everything you need to build a static executable that can play a sound without dealing with installing dependencies on the Roborock (newer firmwares come without APT and it's a mess), and just keep the playSound function called by your MQTT client. Let me know if I can be of help, or post videos of it working ;)

**Reply by charliesz** [reply]
Late report: I tried to get it working but I think my roborock S7 is not powerful enough to handle even oucher. At some point my robot just rebooted and made a factory reset that I had to root my robot again. Luckily I had soldered some cables to the necessary pins on the mainboard, that I just had do disassamble till the small wimple engine is removed.

I just gave up with the s7, maybe I will get back to the project with a retired s5.
